src/Core/Case/CaseBuilder.idr

```diff
@@ -1226,16 +1227,90 @@ simpleCase fc phase fn ty def clauses
          defs <- get Ctxt
          patCompile fc fn phase ty ps def
 
-findReached : CaseTree ns -> List Int
-findReached (Case _ _ _ alts) = concatMap findRAlts alts
+mutual
+  findReachedAlts : CaseAlt ns' -> List Int
```

Comment:
Should this be producing a set instead? Especially given you seem to `nub` the list in the end.

Reply:
This particular function's results are not always nubbed; it's important that reached indices are duplicated where they are used in the list difference below. 

That said, I actually didn't notice we had a set implementation in the compiler libraries (only looked at base), and I _can_ use a sorted set in the implementation of the rep/droprep helpers below so I'll do that. The result of those helpers is indeed always nubbed.